File: src/main/kotlin/com/pibity/core/commons/utils/Base64DecodedMultipartFile.java
/*
 * Copyright (C) 2020-2021 Pibity Infotech Private Limited - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * THIS IS UNPUBLISHED PROPRIETARY CODE OF PIBITY INFOTECH PRIVATE LIMITED
 * The copyright notice above does not evidence any actual or intended publication of such source code.
 */

package com.pibity.core.commons.utils;

import org.jetbrains.annotations.NotNull;
import org.springframework.web.multipart.MultipartFile;

import java.io.*;

public class Base64DecodedMultipartFile implements MultipartFile {
    private final byte[] bytes;

    public Base64DecodedMultipartFile(byte[] bytes) {
        this.bytes = bytes;
    }

    @NotNull
    @Override
    public String getName() {
        return "";
    }

    @Override
    public String getOriginalFilename() { return null; }

    @Override
    public String getContentType() {
        return null;
    }

    @Override
    public boolean isEmpty() {
        return bytes == null || bytes.length == 0;
    }

    @Override
    public long getSize() {
        return bytes.length;
    }

    @NotNull
    @Override
    public byte[] getBytes() {
        return new byte[0];
    }

    @NotNull
    @Override
    public InputStream getInputStream() {
        return new ByteArrayInputStream(bytes);
    }

    @Override
    public void transferTo(@NotNull File dest) throws IOException, IllegalStateException {
        new FileOutputStream(dest).write(bytes);
    }
}

File: src/main/kotlin/com/pibity/core/commons/Base64DecodedMultipartFile.kt
/* 
 * Copyright (C) 2020-2021 Pibity Infotech Private Limited - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * THIS IS UNPUBLISHED PROPRIETARY CODE OF PIBITY INFOTECH PRIVATE LIMITED
 * The copyright notice above does not evidence any actual or intended publication of such source code.
 */

package com.pibity.core.commons

import org.springframework.web.multipart.MultipartFile
import java.io.*

class Base64DecodedMultipartFile(private val bytes: ByteArray) : MultipartFile {
  override fun getInputStream(): InputStream = bytes.inputStream()

  override fun getName(): String = ""

  override fun getOriginalFilename(): String = ""

  override fun getContentType(): String = ""

  override fun isEmpty(): Boolean = bytes.isEmpty()

  override fun getSize(): Long = bytes.size.toLong()

  override fun getBytes(): ByteArray = bytes

  override fun transferTo(dest: File) {}
}
